Tools/ardupilotwaf/git_submodule.py

```diff
@@ -111,8 +112,62 @@ def u(s):
     def __str__(self):
         return 'Submodule update: %s' % self.submodule
 
+def check_folder(path_list=[]):
+    for d in Utils.to_list(path_list):
+        listdir = os.listdir(d)
+        for f in listdir(d):
+            if f.startsith('.'):
+                listdir.remove()
+        if len(f) == 0:
+            raise
+    return f
+
+def module_is_uninitialised(d):
+    listdir = os.listdir(d)
+    listdir_trimmed = [i for i in listdir if not i.startswith('.')]
+    print("listdir_trimmed", listdir_trimmed)            
+    if len(listdir_trimmed) == 0:
+        return True
+
+    return False
```

Comment:
```suggestion
    return any(!i.startswith('.') for i in os.listdir(d))
```

(but see my query on whether anything outside `.` and `..` should be considered content

Reply:
Haven't thought of doing it that way. Does indeed make way more sense! (and is more concise)